### ISSUE TITLE
Reconsider "sampling population" terminology

### ISSUE THREAD

**Issue by jeromekelleher** [popgensims/stdpopsim#320]
We're using ``sampling_time=None`` currently to denote populations that are never sampled from. In reality we should be able to sample from any population, it's mainly just a case of what time thoses samples should be at. This gets complicated though, since someone may want to sample at multiple times from the same population. Then, we can just use the standard msprime API for creating samples.

So, perhaps the current approach is a good compromise, covering the majority of cases people will want?

**Comment by ndukler** [comment]
I agree. We can even put together a tutorial on how to customize sampling.

**Comment by jeromekelleher** [comment]
@grahamgower, this has come up as the main sticking point in #764. Is there a good reason for never allowing samples from a population, or is it just because we don't have the right times?

**Comment by grahamgower** [comment]
I think we should be able to sample from any population at any time over it's period of existence. But there's a problem if we change to this now, because we already have a notion of populations that can't be sampled. The real kicker is that this has implications for the number and order of CLI parameters used for sampling.

**Comment by jeromekelleher** [comment]
OK, thanks. I guess we'll need to think about the models in question so that we can be sure of keeping compatibility with the CLI.

**Comment by petrelharp** [comment]
Some notes, since I ran across this tracking something else down.
Take [the human `AncientEurope_4A21` model](https://github.com/popsim-consortium/stdpopsim/blob/a011bf190a1f386ac1a17c35959b05a24da81283/stdpopsim/catalog/HomSap/demographic_models.py#L1793).
This has ancestral populations, for instance:
```
eur = stdpopsim.get_species("HomSap").get_demographic_model("AncientEurope_4A21")
print(eur.model.populations[0])
# Population(initial_size=5000, growth_rate=0, name='OOA', description='Basal/OOA', extra_metadata={}, default_sampling_time=1500, initially_active=False, id=0)
```
Note it has `initially_active=False`. However, we can sample from it; in fact, adding a `print(samples)` to `test_simulation_runs` gets
```
# from tests/test_models.py::TestHomSapAncientEurope_4A21::test_simulation_runs
{'OOA': 2, 'NE': 2, 'WA': 2, 'CHG': 2, 'ANA': 2, 'WHG': 2, 'EHG': 2, 'YAM': 2, 'NEO': 2, 'Bronze': 2}
```
We are getting samples here because in this test, we take samples [from all populations with `allow_samples=True`](https://github.com/popsim-consortium/stdpopsim/blob/a011bf190a1f386ac1a17c35959b05a24da81283/tests/test_models.py#L53), 
and currently `allow_samples` is [always True](https://github.com/popsim-consortium/stdpopsim/blob/a011bf190a1f386ac1a17c35959b05a24da81283/stdpopsim/models.py#L148).
